### PR TITLE
Bigtable: Add retry parameter to 'table.read_row()'.

### LINKED ISSUE #7955: bigtable: Implement consistent retry mechanisms
**Is your feature request related to a problem? Please describe.**
It is frustrating to utilize a provided retry mechanism for one API call, and being forced to implement our own retry mechanism for a very similar API call because retries aren't provided.

https://googleapis.github.io/google-cloud-python/latest/bigtable/table.html#google.cloud.bigtable.table.Table.read_row

https://googleapis.github.io/google-cloud-python/latest/bigtable/table.html#google.cloud.bigtable.table.Table.read_rows

 **Describe the solution you'd like**
`table.Table.read_rows` accepts a Retry, but `table.Table.read_row` does not. It would be nice if `table.Table.read_row` accepted the same retry mechanism.
 
**Describe alternatives you've considered**
In the absence of a retry mechanism built into read_row, we have written our own.

### PR DESCRIPTION
**Pull request by elisheva-qlogic**
Fix #7955

### CODE REVIEW

File: bigtable/tests/unit/test_table.py [hunk 1]
```diff
@@ -563,6 +563,91 @@ def test_read_row_complete(self):
         column.append(Cell.from_pb(chunk))
         self._read_row_helper(chunks, expected_result, app_profile_id)
 
+    def test_read_row_retry(self):
```
Comment (sduskis) [inline bigtable/tests/unit/test_table.py]: This test should be something like `test_read_row_custom_retry`, and should only test that the `retry_read_row` is passed correctly from `read_row` to `read_rows`.  There's a lot of other stuff in this method that is "orthogonal" to the specific thing you're trying to test, and therefore should be removed; streamlining the test will allow future reviewers to quickly understand the intent of the test.

File: bigtable/tests/unit/test_table.py [hunk 2]
```diff
@@ -563,6 +563,30 @@ def test_read_row_complete(self):
...
         self._read_row_helper(chunks, expected_result, app_profile_id)
 
+    def test_read_row_custom_retry(self):
+        from google.api_core import retry
+        from google.api_core.retry import if_exception_type
+        from google.cloud.bigtable.table import _BigtableRetryableError
+
+        credentials = _make_credentials()
+        client = self._make_client(
+            project="project-id", credentials=credentials, admin=True
```
Comment (sduskis) [inline bigtable/tests/unit/test_table.py]: nit: please don't set `admin=True`.  Ideally, the client / connections should be closed, although I'm not sure we do that consistently.

Comment (elisheva-qlogic) [reply bigtable/tests/unit/test_table.py]: I removed admin=True from this test. All other tests in the file set admin=True when creating the client, should I create a separate PR and remove that from all the other tests? 
Regarding closing the client, none of the other tests do that

Comment (sduskis) [reply bigtable/tests/unit/test_table.py]: It would be nice to have a separate PR for `admin=True` and closing the clients.

**Comment by mf2199** [comment]
This PR is no longer maintained, please refer to [#8490](https://github.com/googleapis/google-cloud-python/pull/8490) instead.

**Pull request closed without merging**